### ISSUE TITLE
Comma merging with '+' sign only on first rule to be concatenated

### ISSUE THREAD

**Issue by ldetomi** [less/less.js#2576]
Today, I faced an issue described in detail in [this topic](http://stackoverflow.com/questions/29912745/less-concatenate-multiple-background-rules/) on StackOverflow.

Solution provided by a user is useful but in my opinion needed workaround is not so efficient in order to achieve desired behaviour.

Please, can you explain me why `+` sign must be placed after both rules that have to be concatenated and only after the first one? In my opinion it seems redundant. I think that this code could be the best one to achieve desired result:

```
.horizontal(@start-color: #555; @end-color: #333; @start-percent: 0%; @end-percent: 100%) {
    background:@start-color;
    background-image: -webkit-linear-gradient(left, @start-color @start-percent, @end-color @end-percent);
    background-image: -o-linear-gradient(left, @start-color @start-percent, @end-color @end-percent);
    background-image: linear-gradient(to right, @start-color @start-percent, @end-color @end-percent);
    background-repeat: repeat-x;
  }

.foo
{
  background-image+:url('bg.png');
  .horizontal;
}
```

Parser encounter `+` sign after FIRST `background-image` in `.foo` declaration and this should mean: "_please, concatenate this `background-image:url('bg.png')` rule with others that start with the same propery, so `background-image`._"

Can you consider this change in behaviour? In case not, why do you think that actual one is the best?

Thank you.

**Comment by lukeapage** [comment]
There was quite some discussion when the feature was introduced. The problem is

```
background: fallback1;
background: fallback2;
background+: ...;
background+: ...;
```

you might want fallbacks to stay unmerged.

There are quite a few scenarios people might be in, depending on how they write their code where different behaviour might be better. We didn't consider or intend property merging to be used for the use-case you are discussing (a parameter to `.horizontal` seems like the easiest solution)

**Comment by ldetomi** [comment]
My idea was based on considerations about "incremental declarations". I think that is much more intuitive for a user, adding rules, while using a total different (and in the meantime, specific) mixin in order to achieve desired behaviour. I thought that "+" sign was introduced with this same feeling in mind...

It seems easiest simply adding a gradient mixin to an existing background image declaration, in case a develioper needs it, instead of be force to totally replace the original background declaration with a very specific mixin. In other words: "if you want only a background image, simply use the omonime rule. If tomorrow, you would like to add ALSO a gradient, you must delete original declaration and replace with a parameter passed to a specific mixin"........ mhmmmmm not so streamlined, in my opinion....

**Comment by ldetomi** [comment]
Continuing to investigate on "comma merging" flexibility, today I asked on SO [another similar question](http://stackoverflow.com/questions/29937026/concatenate-mixins-call-with-comma-merging) with another use case about concatenation of Mixin calls.
Today, solution suggested was simpler and in my opinion perfectly acceptable for the specific case, but availability of `+` also on Mixin Call, could be very useful.

**Comment by matthew-dean** [comment]
I'm not sure if it should be placed here or separately, but while I agree with merging vs. leaving other properties, I consider the output location of property concatenation to be a bug. See: https://github.com/less/less.js/issues/2564#issuecomment-95268231

@lukeapage - Do you think this should be opened as a separate issue? This issue is written as "only on first rule", so it could fall under the same area of "unexpected output", in that the concatenation is placed on the first rule and not the last.

**Comment by seven-phases-max** [comment]
Closing due to inactivity. (The separate and unrelated https://github.com/less/less.js/issues/2564#issuecomment-95268231 needs its own ticket).
